crates/task-impls/src/quorum_vote.rs

```diff
@@ -70,12 +74,168 @@ struct VoteDependencyHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     view_number: TYPES::Time,
     /// Event sender.
     sender: Sender<Arc<HotShotEvent<TYPES>>>,
+    /// The current version of HotShot
+    version: Version,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> VoteDependencyHandle<TYPES, I> {
+    async fn update_shared_state(
```

Comment:
cc @bfish713 and @shenkeyao please check this against the old logic. It *looks* right, but I could have goofed.